### ISSUE TITLE
Can't use 'defined(@array)'

### ISSUE THREAD

**Issue by moolen** [kentcdodds/cloc#1]
Hi kentcdodds,
I'm running Arch Linux with perl 5.22; Installing your cloc package via npm results in a lot of deprecation warnings and errors due to my perl version.

```
Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /home/moolen/.npm/bin/cloc line 1284.
```

I found this specific bug that matches my error case:
http://sourceforge.net/p/cloc/bugs/135/

I installed the current version from the Official repository at https://github.com/AlDanial/cloc and it seems that the bugs have been fixed there.

Could you please update the cloc script or point to the official upstream repo?

Best,
moolen

**Comment by supersonicclay** [comment]
+1 I am getting the same with cloc 1.0.4 and Windows 7 64bit

**Comment by WangJi** [comment]
+1
They are both in version 1.04.
That's weird.

**Comment by wald-tq** [comment]
You are using an old Perl version. I got this error too. On Windows you are probably using the perl version shipped with msysgit, which is also perl 5.22.

You need at least perl Version 5.6 or higher. See: https://github.com/AlDanial/cloc I am looking for a resolution now.

Edit: perl 5.22 is higher than 5.6 of course. It should work.

Resolution: There is a bug (https://bugs.archlinux.org/task/45494) included in the bugreport is a patch which you can apply against the cloc Version 1.64:

```
patch -p1 < cloc.1.64.patch
```

After that it works.

**Comment by kentcdodds** [comment]
I need to give you an apology. For some reason, I want watching this repository so I want getting notifications about issues and pull requests. I am so incredibly sorry. Please forgive me. I really feel awful. I hope my unresponsiveness doesn't hinder you contributing more in the future.

As for your specific issue, I'll give it a closer look soon.

**Comment by wald-tq** [comment]
No need to apologize. Problem should be resolved with the Release 2.0.0 which includes cloc 1.66.

**Comment by kentcdodds** [comment]
Let me know! But I'll go ahead and close this now.
